src/infrastructure/initialize.hpp

```diff
@@ -17,6 +17,15 @@
 #include "mpi.h"
 
 namespace serac {
+
+/**
+ * @brief Returns the number of processes and rank for an MPI communicator
+ * @param comm The MPI communicator to initialize with
+ * @return A pair containing the size and rank relative to the provided MPI communicator
+ * @pre The logger must be initialized (to display error messages)
+ */
+std::pair<int, int> getMPIInfo(MPI_Comm comm = MPI_COMM_WORLD);
```

Comment:
The `@return` comment implies that `std::get<0>(getMPIInfo(...))` is the size and `std::get<1>(getMPIInfo(...))` the rank, but I had to check the implementation to be sure. Maybe we could explicitly state

```
// A pair containing {size, rank} relative to the provided MPI communicator
```

or returning a self-documenting struct like

```
struct MPIInfo{ int size; int rank; };
```